src/segment_language.go

```diff
@@ -43,7 +65,17 @@ func (l *language) string() string {
 		return l.props.getString(MissingCommandTextProperty, MissingCommandText)
 	}
 	l.setVersion()
-	return l.version
+
+	// if regex found no version
+	if l.version.full == "" {
```

Comment:
this is an internal property only, removing the hyperlink creation in `version.string()` will allow you to not do this check at all. On top of that, it's not even useful as you return the value you validate it contains meaning the last return does exactly the same.

Reply:
I don't get those points related to the generation of the hyperlink. In the end, to build the hyperlink all properties from version are needed. If it's not done from a function on `version`, then I need to get the data from those properties inside `language.string()`?